src/directory/mmap_directory.rs

```diff
@@ -192,17 +192,19 @@ impl MmapDirectory {
     /// exist or if it is not a directory.
     pub fn open<P: AsRef<Path>>(directory_path: P) -> Result<MmapDirectory, OpenDirectoryError> {
         let directory_path: &Path = directory_path.as_ref();
-        if !directory_path.exists() {
+        let canonical_path: PathBuf = directory_path.canonicalize().map_err(
+            |io_err| OpenDirectoryError::wrap_io_error(io_err, PathBuf::from(directory_path))
+        )?;
+        if !canonical_path.exists() {
             Err(OpenDirectoryError::DoesNotExist(PathBuf::from(
```

Comment:
You need the  `return` keyword here.

Reply:
> `we were returning the if {} else {} expression`

Ooooh... Ok I get it, the if block is evaluated so it's value is returned. Thx, my "rust by example" is rusty.